app/jobs/set_tags_job.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+##
+# job to set tags on objects
+class SetTagsJob < GenericJob
+  queue_as :default
+
+  DELIM = "\t"
+
+  def perform(bulk_action_id, params)
+    super
+
+    with_bulk_action_log do |log_buffer|
+      log_buffer.puts("#{Time.current} Starting #{self.class} for BulkAction #{bulk_action_id}")
+      update_druid_count
+
+      pids.map { |druid_with_tag| druid_with_tag.split(DELIM) }.each do |pid|
+        log_buffer.puts("#{Time.current} #{self.class}: Attempting to set tags for #{pid[0]} (bulk_action.id=#{bulk_action_id})")
+        set_tags(pid[0], pid[1], log_buffer)
```

Comment:
Isn't this dropping all but the first tag? I ask because I'm not sure what the value of `pids` is on line 17. Is it just whatever the user puts in the textarea (via the default bulk action UI)? In that case, I assume pids looks something like `["druid:1\tTag : One\tTag : Two", "druid:2\tTag : 2\tTag : 3"]`

That means after the `#map` operation, the value of `pid` in the `#each` block looks like `["druid:1", "Tag : One", "Tag : Two"]` and then `["druid:2", "Tag : 2", "Tag : 3"]`. So `pid[0]` will be the druid and `pid[1]` will be the first tag. The other tags are ignored.

Would you consider something like:

```ruby
# ...
pids.map { |druid_with_tag| druid_with_tag.split("\t") }.each do |druid, *tags| 
  log_buffer.puts("#{Time.current} #{self.class}: Attempting to set tags for #{druid} (bulk_action.id=#{bulk_action_id})")
  set_tags(druid, log_buffer, *tags)
end
# ...

private

def set_tags(druid, log_buffer, *tags)
# ...
end
```

:question:

Reply:
It is (at the moment) - I made a comment in slack that I needed to address this issue, yes. It's incoming ;-)